### PR TITLE
React to ContextAccessor changes

### PR DESCRIPTION
**Pull request by HaoK**
- HttpContextAccessor
- IScopeLocal

cc @yishaigalatzer @rynowak

### CODE REVIEW

**Comment by yishaigalatzer** [comment]
I chatted earlier with @davidfowl and he wasn't convinced that this is the right path yet. So please chat with him first before proceeding.

File: src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ private static void ConfigureDefaultServices(IServiceCollection services, IConfi
             services.AddOptions(configuration);
             services.AddDataProtection(configuration);
             services.AddRouting(configuration);
-            services.AddContextAccessor(configuration);
+            services.AddScopeLocal(configuration);
```
Comment (divega) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: Am I reading this correctly and AddMvc() stick the IConfiguration into DI? I thought that was not supposed to happen.

Comment (HaoK) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: You aren't reading this correctly, this is the extension method that is adding IScopeLocal as a scoped service to DI...

**Comment by HaoK** [comment]
Fowler is ok, he was just concerned about potential naming, do the changes look ok otherwise, ignoring the name for now?

File: src/Microsoft.AspNet.Mvc/Properties/debugSettings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "Profiles": []
```
Comment (yishaigalatzer) [inline src/Microsoft.AspNet.Mvc/Properties/debugSettings.json]: this change looks unrelated.

**Comment by yishaigalatzer** [comment]
:shipit: when @davidfowl is happy with the name consolidation.

File: src/Microsoft.AspNet.Mvc.Core/MvcRouteHandler.cs [hunk 3]
```diff
@@ -95,27 +95,25 @@ public string GetVirtualPath([NotNull] VirtualPathContext context)
             var optionsAccessor = services.GetRequiredService<IOptions<MvcOptions>>();
             actionContext.ModelState.MaxAllowedErrors = optionsAccessor.Options.MaxModelValidationErrors;
 
-            var contextAccessor = services.GetRequiredService<IContextAccessor<ActionContext>>();
-            using (contextAccessor.SetContextSource(() => actionContext, PreventExchange))
+            var contextAccessor = services.GetRequiredService<IScopeLocal<ActionContext>>();
```
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Core/MvcRouteHandler.cs]: Can `PreventExchange` be deleted here?

File: src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs (hunk 1 above)
Comment (divega) [inline src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs]: D'oh! Makes sense now.

**Comment by HaoK** [comment]
Updated with new name: IScopedInstance

**Comment by HaoK** [comment]
Merged 157b63375829682be1f3f25c270c2c3b019e9833
